cocos/scripting/js-bindings/manual/jsb_cocos2dx_extension_manual.cpp

```diff
@@ -132,6 +134,180 @@ static bool js_cocos2dx_extension_loadRemoteImage(se::State& s)
+    {
+        bool success = false;
+
+        Image* image = new (std::nothrow) Image();
+        if (image->initWithImageData(data.data(), data.size()))
+        {
+            if (texture->initWithImage(image))
+            {
+                success = true;
+            }
+            else
+            {
+                CCLOGERROR("js_cocos2dx_extension_loadRemoteImageOn: Failed to initWithImage.");
+            }
+        }
+        CC_SAFE_RELEASE_NULL(image);
+
+        onCallback(success);
+        Director::getInstance()->getScheduler()->performFunctionInCocosThread([downloader](){
+            delete downloader;
+        });
+    };
+
+    downloader->onTaskError = [=](const cocos2d::network::DownloadTask& task, int errorCode, int errorCodeInternal, const std::string& errorStr)
+    {
+        onCallback(false);
+        Director::getInstance()->getScheduler()->performFunctionInCocosThread([downloader](){
+            delete downloader;
+        });
+    };
+
+    downloader->createDownloadDataTask(url);
+    return true;
+}
+SE_BIND_FUNC(js_cocos2dx_extension_initRemoteImage)
+
+static ThreadPool* _threadPool = nullptr;
+static EventListenerCustom* _resetThreadPoolListener = nullptr;
+static ThreadPool* getThreadPool()
+{
+    if (_threadPool == nullptr)
+    {
+        _threadPool = ThreadPool::newSingleThreadPool();
+        _resetThreadPoolListener = Director::getInstance()->getEventDispatcher()->addCustomEventListener(Director::EVENT_RESET, [](cocos2d::EventCustom*){
+            CC_SAFE_DELETE(_threadPool);
+            cocos2d::Director::getInstance()->getEventDispatcher()->removeEventListener(_resetThreadPoolListener);
+            _resetThreadPoolListener = nullptr;
+        });
+    }
+    return _threadPool;
+}
+
+static bool js_cocos2dx_extension_initTextureAsync(se::State& s)
+{
+    const auto& args = s.args();
+    int argc = (int)args.size();
+    if (argc != 3)
+    {
+        SE_REPORT_ERROR("wrong number of arguments: %d, was expecting %d", argc, 3);
+        return false;
+    }
+
+    bool ok = false;
+
+    // get texture
+    cocos2d::Texture2D* texture = nullptr;
+    ok = seval_to_native_ptr(args[0], &texture);
+    SE_PRECONDITION2(ok, false, "Converting 'texture' failed!");
+
+    // get url
+    std::string url;
+    ok = seval_to_std_string(args[1], &url);
+    SE_PRECONDITION2(ok, false, "Converting 'url' failed!");
+
+    // get callback
+    se::Value func = args[2];
+    assert(func.isObject() && func.toObject()->isFunction());
+    func.toObject()->root();
+
+    auto onCallback = [=](bool success){
+        se::ScriptEngine::getInstance()->clearException();
+        se::AutoHandleScope hs;
+
+        se::ValueArray args;
+        args.resize(1);
+
+        args[0].setBoolean(success);
+        func.toObject()->call(args, nullptr);
+    };
+
+    getThreadPool()->pushTask([=](int tid){
+        bool success = false;
+
+        Data data = FileUtils::getInstance()->getDataFromFile(url);
```

Comment:
getDataFromFile isn't thread safe.
you could try to get the full path of url before pushTask, and use `lambda` to capture the std::string of full path.